src/native/eventpipe/ep-session.c

```diff
@@ -100,28 +100,44 @@ EP_RT_DEFINE_THREAD_FUNC (streaming_thread)
 	session->streaming_thread = thread_params->thread;
 
 	bool success = true;
-	ep_rt_wait_event_handle_t *wait_event = ep_session_get_wait_event (session);
 
 	ep_rt_volatile_store_uint32_t (&session->started, 1);
 
 	EP_GCX_PREEMP_ENTER
-		while (ep_session_get_streaming_enabled (session)) {
-			bool events_written = false;
-			if (!ep_session_write_all_buffers_to_file (session, &events_written)) {
-				success = false;
-				break;
-			}
+		if (ep_session_type_uses_buffer_manager (session->session_type)) {
+			ep_rt_wait_event_handle_t *wait_event = ep_session_get_wait_event (session);
+			while (ep_session_get_streaming_enabled (session)) {
+				bool events_written = false;
+				if (!ep_session_write_all_buffers_to_file (session, &events_written)) {
+					success = false;
+					break;
+				}
+
+				if (!events_written) {
+					// No events were available, sleep until more are available
+					ep_rt_wait_event_wait (wait_event, EP_INFINITE_WAIT, false);
+				}
 
-			if (!events_written) {
-				// No events were available, sleep until more are available
-				ep_rt_wait_event_wait (wait_event, EP_INFINITE_WAIT, false);
+				// Wait until it's time to sample again.
+				const uint32_t timeout_ns = 100000000; // 100 msec.
+				ep_rt_thread_sleep (timeout_ns);
 			}
+		} else if (session->session_type == EP_SESSION_TYPE_USEREVENTS) {
+			// User events session, write all user events tracepoints to the file.
+			while (ep_session_get_streaming_enabled (session)) {
+				EP_ASSERT (session->continuation_stream != NULL);
+				if (ep_ipc_continuation_stream_connection_closed (session->continuation_stream)) {
+					success = false;
+					break;
+				}
 
-			// Wait until it's time to sample again.
-			const uint32_t timeout_ns = 100000000; // 100 msec.
-			ep_rt_thread_sleep (timeout_ns);
+				// Wait until it's time to sample again.
```

Comment:
```suggestion
				// Wait until it's time to poll again.
```
I see you copied it from above but 'sample' seems like an odd description of this :)